test/MyTested.AspNetCore.Mvc.Authentication.Test/BuildersTests/ViewComponentsTests/ViewComponentBuilderTests.cs

```diff
@@ -63,5 +64,104 @@ public void WithAuthenticatedUserShouldPopulateProperUserWhenUserWithUserBuilder
                 });
         }
         
+        [Fact]
+        public void WithAuthenticatedUserShouldPopulateUserRoleProperly()
+        {
+            MyViewComponent<UserComponent>
+                .Instance()
+                .WithUser(new List<string>
+                {
+                    "Administrator"
+                })
+                .InvokedWith(c => c.Invoke())
+                .ShouldReturn()
+                .View()
+                .ShouldPassForThe<UserComponent>(viewComponent =>
+                {
+                    var user = viewComponent.User as ClaimsPrincipal;
+
+                    Assert.NotNull(user);
+                    Assert.True(user.IsInRole("Administrator"));
+                    Assert.Equal("Passport", user.Identity.AuthenticationType);
+                    Assert.True(user.Identity.IsAuthenticated);
+                });
+        }
+
+        [Fact]
+        public void WithAuthenticatedUserShouldPopulateUserNameAndRoleProperly()
+        {
+            MyViewComponent<UserComponent>
+                .Instance()
+                .WithUser("NewUserName",
```

Comment:
Fix these parameters. They should be like:
```
                .WithUser("NewUserName", new List<string>
                {
                      "Administrator"
                })
```

Reply:
Parameters are fixed.